### ISSUE TITLE
fdo: Let CMake check whether wl_egl_create_window() can be linked

### ISSUE THREAD

**Issue by aperezdc** [Igalia/cog#39]
After #38 it can happen that the checks succeed, yet building the FDO platform module won't be possible due `wayland-egl`. It would be interesting to use [CHECK_FUNCTION_EXISTS()](https://cmake.org/cmake/help/v3.3/module/CheckFunctionExists.html) to make sure that the needed functions can be linked in. Probably checking for `wl_egl_create_window()` is enough — I would expect that if that is usable, the other ones will be available as well.

**Comment by aperezdc** [comment]
FWIW, for those curious about why this is something that would be nice to have, make sure to read the comments from PR #38, in particular [this one](https://github.com/Igalia/cog/pull/38#issuecomment-399075609).

The underlying idea is that it is nicer (and clearer to users) to stop at CMake configuration time and provide a meaningful message than letting the build fail later on.

**Comment by philn** [comment]
This can be closed maybe? I see meson doing `cxx.has_function('wl_egl_window_create', dependencies: dep)` on wayland-egl and egl deps.

**Comment by aperezdc** [comment]
> This can be closed maybe? I see meson doing `cxx.has_function('wl_egl_window_create', dependencies: dep)` on wayland-egl and egl deps.

Indeed! I forgot we had this issue to track this, let's close it.
